plugin/grpctrace/interceptor_test.go

```diff
@@ -0,0 +1,127 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package grpctrace
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"google.golang.org/grpc"
+
+	"go.opentelemetry.io/otel/api/global"
+	export "go.opentelemetry.io/otel/sdk/export/trace"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+)
+
+type testExporter struct {
+	spanMap map[string][]*export.SpanData
+}
+
+func (t *testExporter) ExportSpan(ctx context.Context, s *export.SpanData) {
+	t.spanMap[s.Name] = append(t.spanMap[s.Name], s)
+}
+
+type mockCCInvoker struct {
+	ctx context.Context
+}
+
+func (mcci *mockCCInvoker) invoke(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, opts ...grpc.CallOption) error {
+	mcci.ctx = ctx
+	return nil
+}
+
+type mockProtoMessage struct {
+}
+
+func (mm *mockProtoMessage) Reset()         {}
+func (mm *mockProtoMessage) String() string { return "mock" }
+func (mm *mockProtoMessage) ProtoMessage()  {}
+
+func TestUCIFullyQualifiedMethodNameSetsServiceNameAttribute(t *testing.T) {
+
+	expectedServiceName := "serviceName"
+	fullyQualifiedName := fmt.Sprintf("/foo.%s/bar", expectedServiceName)
+
+	testUCISetsExpectedNameAttribute(t, fullyQualifiedName, expectedServiceName)
+}
+
+func TestUCISimpleMethodNameSetsServiceNameAttribute(t *testing.T) {
+	expectedServiceName := "serviceName"
+	simpleName := fmt.Sprintf("/%s/bar", expectedServiceName)
+
+	testUCISetsExpectedNameAttribute(t, simpleName, expectedServiceName)
+}
+
+func TestUCIInvalidMethodSetsEmptyNameAttribute(t *testing.T) {
+	expectedServiceName := ""
+	emptyName := "invalidMethodName"
+
+	testUCISetsExpectedNameAttribute(t, emptyName, expectedServiceName)
+}
+
+func TestUCILongMethodNameSetsServiceNameAttribute(t *testing.T) {
+	expectedServiceName := "serviceName"
+	emptyServiceFullName := fmt.Sprintf("/github.com.foo.baz.%s/bar", expectedServiceName)
+
+	testUCISetsExpectedNameAttribute(t, emptyServiceFullName, expectedServiceName)
+}
+
+func TestUCINonAlhanumericMethodNameSetsServiceNameAttribute(t *testing.T) {
+	expectedServiceName := "serviceName_123"
+	emptyServiceFullName := fmt.Sprintf("/faz.bar/baz.%s/bar", expectedServiceName)
+
+	testUCISetsExpectedNameAttribute(t, emptyServiceFullName, expectedServiceName)
+}
```

Comment:
It looks like these could all be collapsed into a single table-driven test since they're ultimately all using the `testUCISetsExpectedNameAttribute()` helper.  Something like this:

```go
type nameAttrTestCase struct{
  testName string,
  expectedServiceName string,
  fullNameFmt string,
}

func TestUCINameAttribute(t *testing.T) {
  for _, tc := range []nameAttrTestCase{
    {
      testName: "Fully Qualified Method Name",
      expectedServiceName: "serviceName",
      fullNameFmt: "/foo.%s/bar",
    },
  } {
    testUCISetsExpectedNameAttribute(t, tc)
  }
}
```

With corresponding changes to the helper to pull the fields out of the inbound struct, use 
```go
expected := fmt.Sprintf(tc.fullNameFmt, tc.expectedServiceName)
```
etc.

Reply:
Done. I used golang's subtest to do it: https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks
It makes the logs nicer, as you can see exactly what cases are running.